cmd/lvmd/app/root.go

```diff
@@ -108,13 +111,30 @@ func subMain(ctx context.Context) error {
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 	defer stop()
 
+	var pprofServer *http.Server
+	if config.ProfilingBindAddress != "" {
+		pprofServer = profiling.NewProfilingServer(config.ProfilingBindAddress)
+		go func() {
+			if err := pprofServer.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+				logger.Error(err, "pprof server error")
+			}
+		}()
+	}
+
 	go func() {
 		ticker := time.NewTicker(10 * time.Minute)
 		for {
 			select {
 			case <-ctx.Done():
 				ticker.Stop()
 				grpcServer.GracefulStop()
+				if pprofServer != nil {
+					shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
This is the wrong background context and also the shutdown should be bound to the base context

Reply:
Since the `ctx` has been closed, we can not use it, IMO.

Reply:
Hmm, we already call `grpcServer.GracefulStop`, the `subMain` may exit before shutting down pprofServer. It would be better to care about this case or close the server immediately.

Reply:
@jakobmoellerdev I believe I understand what you're asking for now, and have made the changes. Lmk, thanks.